tests/test_filters.py

```diff
@@ -361,3 +368,19 @@ def test_power_filter_safeguard(dtype, caplog):
     # Check caplog for warning.
     msg = f"setting {num_eps} extremal filter value(s) to zero."
     assert msg in caplog.text
+
+
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_array_filter_dtype_passthrough(dtype):
+    """
+    Do to a bug in scipy versions < 1.10.1, scipy's interpolator crashes
```

Comment:
Yesterday in our dev meeting I suggested you double check if this was specific to <1.10.1 and if so determine whether we can just bump the requirements up to the next version.  Was the return type still an issue in later versions too or caused some other conflict? (If so update the comment, if not we should be able to remove the workaround). Thanks

Reply:
Ok, looked into this. It was a 1.10.0 bug. It was fixed in the sense that it no longer crashes when using singles, but singles are upcast somewhere internal to scipy so we'll need to keep the recasting.

I will bump the version and remove our own upcast, leaving a comment that explains the recast.

Reply:
I see. Bummer.  Yeah, I figured you did look into it and something wasn't going to work. Sounds great.

Reply:
done